fxapom/fxapom.py

```diff
@@ -54,6 +58,29 @@ def create_account(self):
         self.session.verify_email_code(m["headers"]["x-verify-code"])
         return self
 
+    def delete_account(self):
+        try:
+            self.account.clear()
+            self.client.destroy_account(self.email, self.password)
+        except ClientError as err:
+            # 'Unknown Account' error is ok - account already deleted
+            # https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#response-format
+            if err.errno == 102:
+                return
+            raise
+
+    def login(self):
+        try:
+            session = self.client.login(self.email, self.password)
+            return session
+        except ClientError as err:
+            # 'Unknown Account' error is the only one we care about and will
+            # cause us to throw a custom exception
+            # https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#response-format
+            if err.errno == 102:
+                raise AccountNotFoundException('FxA Account Not Found')
```

Comment:
The `ClientError` looks like:

`ClientError: Unknown account`

The issue is that there can be many different types of exceptions that are all `ClientError`, so simply looking for that exception type in the test wouldn't work. I thought it better to catch `ClientError` in login, and if it is the specific exception that we care about (which is `errno` 102) then to raise my own custom exception, which I can then catch in the test. The test in question is `test_delete_account` and it wants `login` to fail with that specific exception.

I thought this better than adding the logic that looks for `errno == 102` into the test itself.
Does that make sense?